tests/integration/test_dbapi_integration.py

```diff
@@ -362,6 +376,22 @@ def test_transaction_multiple(trino_connection_with_transaction):
     assert len(rows2) == 1000
 
 
+def test_transaction_autocommit(trino_connection_in_autocommit):
+    with trino_connection_in_autocommit as connection:
+        connection.start_transaction()
+        cur = connection.cursor()
+        cur.execute(
+            """
+            CREATE TABLE memory.default.nation
+            AS SELECT * from tpch.tiny.nation
+            """)
+
+        with pytest.raises(TrinoUserError) as transaction_error:
+            cur.fetchall()
+        assert "Catalog only supports writes using autocommit: memory" \
+               in str(transaction_error.value)
```

Comment:
Would this need to be conditional based on Trino version being tested?

Should we add a fixture called `honors_autocommit` that uses the Trino version being tested to return a boolean that we can use to skip tests that don't apply to certain versions. Similar to JDBC in https://github.com/trinodb/trino/blob/83456f0ee6aeed817d4d0cab477f91f1e5821557/testing/trino-test-jdbc-compatibility-old-driver/src/test/java/io/trino/JdbcDriverCapabilities.java#L79

Reply:
yes, I would need to check when autocommit mechanics was changed. 

I will add a conditional check and skip this test depending on the version.

Reply:
done!